File: weekly_test10/equi_filter.py
#!/usr/bin/python3

import sys

def is_equi_word(word):
	letter_count = {}
	for letter in word.lower():
		letter_count[letter] = letter_count.get(letter, 0) + 1
	return len(set(letter_count.values())) == 1

for line in sys.stdin:
	words = line.split()
	equi_words = filter(is_equi_word, words)
	print(' '.join(equi_words))

File: weekly_test10/equi_filter.pl
#!/usr/bin/perl -w

while ($line = <STDIN>) {
	@words = split /\s+/, $line;
	foreach $word (@words){
		@letters = split //, lc $word;
		%letter_count = ();
		foreach $letter (@letters){
			$letter_count{$letter}++;
		}
		$value1 = 0;
                $equi = 0;
		
		foreach $value (values %letter_count) {
			if ($value1 == 0){
				$value1 = $value;
			}
			if ($value1 == $value){
				$equi = 1;
			}
			else{
				$equi = 0;
				last;
			}
		}
		if ($equi == 1){
			print "$word ";
		}
	}
	print "\n";
}

